src/pages/index.js

```diff
@@ -1,100 +1,259 @@
-// Imports modules and classes
-import {
-  formValidationSettings,
-  initialCards,
-  selectors,
-} from "../utils/constants.js";
+// Imports
+import Api from "../components/Api.js";
+import { formValidationSettings, selectors } from "../utils/constants.js";
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
-import "../pages/index.css";
+import Popup from "../components/Popup.js"; // Import Popup for the delete confirmation popup
+import "../pages/index.css"; // Import styles for the page
 
-// Destructured selectors from the constants file
+// Initialize the API instance
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "765a56ab-1403-4444-8fac-5fc1871e9695",
+    "Content-Type": "application/json",
+  },
+});
+
+// Destructured selectors from constants
 const {
   profileEditForm,
   addCardForm,
   profileEditButton,
   addCardButton,
   profileTitleInput,
   profileDescriptionInput,
-  cardListEl,
+  avatarEditForm,
 } = selectors;
 
-// Initialize UserInfo to manage and update profile data (name and description)
-const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  descriptionSelector: ".profile__description",
+// Cache DOM elements for buttons to avoid querying the DOM repeatedly
+const saveProfileButton = document.querySelector(
+  "#profile-edit-popup .popup__button"
+);
+const saveCardButton = document.querySelector("#add-card-popup .popup__button");
+const saveAvatarButton = document.querySelector(
+  "#avatar-edit-popup .popup__button"
+);
+const avatarEditButton = document.querySelector("#edit-avatar-button");
+
+// Initialize UserInfo to manage and update profile data on the page
+const userInfoInstance = new UserInfo({
+  nameSelector: ".profile__title", // Selector for profile name
+  descriptionSelector: ".profile__description", // Selector for profile description
+  avatarSelector: ".profile__image", // Selector for profile avatar
 });
 
-// Function to open the profile edit popup with current user data
-function handleProfileEdit() {
-  const userData = userInfo.getUserInfo(); // Get current profile data (name, description)
-  profileTitleInput.value = userData.name; // Pre-fill the form with the current name
-  profileDescriptionInput.value = userData.description; // Pre-fill the form with the current description
+// Initialize the delete confirmation popup using the Popup class
+const deletePopupInstance = new Popup("#delete-card-popup");
```

Comment:
The `Popup` class should not be used in `index.js` because it's only for **extending**